rewritev/Main.hs

```diff
@@ -38,18 +38,43 @@ main = do
     else
         runWithArgs as
 
+finiteArg :: String -> Bool
+finiteArg ('_':_) = True
+finiteArg _ = False
+
+isFlagOrNumber :: String -> Bool
```

Comment:
You can use isDigit here :
https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Char.html#v:isDigit